feat(ast/estree): add decorators to ESTree AST

overlookmotel opened the pull request:
Close #10921.

Add decorators to ESTree AST. `decorators` field is added to `ClassDeclaration`, `ClassExpression`, `PropertyDefinition`, and `MethodDefinition` - as per [ESTree spec](https://github.com/estree/estree/blob/52399d923111fe2a4dcfe1ede795e08356f35ddc/stage3/decorators.md).

Bump `acorn-test262` dependency to include https://github.com/oxc-project/acorn-test262/pull/42 and https://github.com/oxc-project/acorn-test262/pull/43, which add test fixtures for decorators.

overlookmotel commented:
* **#11393** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11393?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11393?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=11393).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **May 31, 1:19 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
feat(ast/estree): add decorators to ESTree AST (#11393)

Close #10921.

Add decorators to ESTree AST. `decorators` field is added to `ClassDeclaration`, `ClassExpression`, `PropertyDefinition`, and `MethodDefinition` - as per [ESTree spec](https://github.com/estree/estree/blob/52399d923111fe2a4dcfe1ede795e08356f35ddc/stage3/decorators.md).

Bump `acorn-test262` dependency to include https://github.com/oxc-project/acorn-test262/pull/42 and https://github.com/oxc-project/acorn-test262/pull/43, which add test fixtures for decorators.
